dev/com.ibm.ws.artifact.zip/src/com/ibm/ws/artifact/zip/internal/ZipFileContainer.java

```diff
@@ -928,13 +966,65 @@ public boolean isChildOf(String parentPath, String childPath) {
 
         try {
             return ZipFileContainerUtils.collectZipEntries(useZipFile);
-
         } finally {
             closeZipFileHandle(); // throws IOException
         }
     }
 
-    //
+    private boolean initializeMultiRelease() {
+        if (isMultiRelease != null)
+            return isMultiRelease;
+
+        // Multi-Release jars only apply to JDK 9+
+        if (CURRENT_JAVA_VERSION < 9) {
+            isMultiRelease = false;
+            return isMultiRelease;
+        }
+
+        // Multi-Release jars can be disabled with JDK system properties
+        if (JDK_DISABLE_MULTI_RELEASE) {
+            if (TraceComponent.isAnyTracingEnabled() && tc.isDebugEnabled())
+                Tr.debug(tc, "JDK system property set to globally disable multi-release jars");
+           isMultiRelease = false;
+           return isMultiRelease;
+        }
+
+        synchronized (multiReleaseLock) {
+            if (isMultiRelease != null)
+                return isMultiRelease;
+
+            // Look for manifest location. Can have various capitalization schemes
+            ZipFileEntry mfEntry = getEntry("META-INF/MANIFEST.MF");
+            if (mfEntry == null)
+                getEntry("META-INF/MANIFEST.mf");
```

Comment:
good catch, yes I was missing the `mfEntry` assignment here. Looking at the JDK code, the `URLClassLoader` uses `JarFile` to load jars, and `JarFile` is able to detect a manifest with any capitalization. Although the spec clearly does state that it must be all caps.

It would be better performance to just check for all caps and move on, but I feel like I've seen `MANIFEST.mf` before, and trying to catch this in a support situation would be quite tricky. So it's a perf vs. "be consistent with the JDK" tradeoff here.

Reply:
My concern is that we should possibly do this in a generic way and if we find a non all caps MANIFEST.MF file we path it to the path MANIFEST.MF so that we don't need to search for it the way this code is and then the other classloader code would automatically get the benefit when they search for META-INF/MANIFEST.MF.  It would be good to get @tbitonti take on it as well.

Reply:
if we implemented the generic approach, could that have an impact on existing applications with invalid manifests?

Reply:
Ok; did some digging, and the TWAS WTP code looks for META-INF/MANIFEST.MF, and doesn't check alternate cases.  It looks like ContainerClassLoader, Liberty, only looks for META-INF/MANIFEST.MF, also without checking alternate cases.  I'm thinking we will be OK to only look for that one value and ignore alternate cases.

Reply:
Although, if URLClassloader accepts alternate cases, that seems to set a precedent.  And I vaguely remember seeing archives with alternate case manifest files.